src/dft.py

```diff
@@ -3,6 +3,12 @@
 from typing import Iterable
 from tree import T
 
+def in_order_L(t: T | None, L: list[int]):
+    if(t!=None):
+        in_order_L(t.left,L)
+        L.append(t.val)
+        in_order_L(t.right,L)
```

Comment:
You _are_ of course using a stack here, but it is the recursion stack. For a deep tree, you will run out of stack space, and the program will crash. That is why the exercise was to use an explicit stack instead.